src/main/java/seedu/mark/logic/parser/TabCommandParser.java

```diff
@@ -0,0 +1,126 @@
+package seedu.mark.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.mark.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import seedu.mark.logic.commands.TabCommand;
+import seedu.mark.logic.commands.TabCommand.Tab;
+import seedu.mark.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new TabCommand object.
+ */
+public class TabCommandParser implements Parser<TabCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the TabCommand
+     * and returns a TabCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public TabCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+
+        return new TabCommand(parseTab(args));
+    }
+
+
+    /**
+     * Parses a {@code String args} into a {@code Tab}.
+     * {@code args} is valid if it is either 1, 2 or 3, or the keywords of tab.
+     * @param args A valid argument of Tab
+     * @return The corresponding tab
+     * @throws ParseException if the given {@code arg} is invalid.
+     */
+    public static Tab parseTab(String args) throws ParseException {
+
+        TabCommand.Tab tab;
+        try {
+            tab = parseTabIndex(args);
+        } catch (ParseException pe_index) {
+
+            try {
+                tab = parseTabKeyword(args);
+            } catch (ParseException pe_kw) {
+                throw pe_index;
+            }
+
+        }
+
+        return tab;
+    }
```

Comment:
Is it better to separate the index parsing and keyword parsing, or merge them together in a single switch?

Reply:
For now it looks okay to me